incubator/hnc/Makefile

```diff
@@ -247,7 +247,32 @@ kind-deploy:
 # Go test suite has a 10m timeout, and the flag disables that timeout (as of
 # July 2020, these tests take ~15m and that number is expected to grow).
 .PHONY: test-e2e
-test-e2e:
+test-e2e: exclude-system-namespaces
+	go clean -testcache
+	go test -v -timeout 0 ./test/e2e/...
+
+# This batch test will run e2e tests N times on the current cluster the user
+# deployed (either kind or a kubernetes cluster), e.g. "make test-e2e-batch N=10"
+.PHONY: test-e2e-batch
+test-e2e-batch: exclude-system-namespaces
+	number=1 ; while [[ $$number -le $N ]] ; do \
+		echo $$number ; \
+    ((number = number + 1)) ; \
+		go clean -testcache ; \
+		go test -v -timeout 0 ./test/e2e/... ; \
+	done
+
+exclude-system-namespaces:
+	@echo
+	@echo "Ensuring all system namespaces are excluded from HNC..."
+	@echo
+	kubectl label ns hnc-system hnc.x-k8s.io/excluded-namespace=true --overwrite
+	@echo
+	kubectl label ns kube-node-lease hnc.x-k8s.io/excluded-namespace=true --overwrite
+	@echo
+	kubectl label ns kube-public hnc.x-k8s.io/excluded-namespace=true --overwrite
+	@echo
+	kubectl label ns kube-system hnc.x-k8s.io/excluded-namespace=true --overwrite
 	@echo
```

Comment:
Any reason you wanted to keep this so verbose?

Reply:
Ah now I can remove the spaces. They were there for easier to read for whether each label is just added.

Reply:
Updated it to make these commands silent (with @).